### ISSUE TITLE
Menu item with submenu does not read out how many items it contains (voiceover)

### ISSUE THREAD

**Issue by maspychaj-sage** [Sage/carbon#4922]
### Current behaviour

When you focus on first element of Navigation Bar Voice Over tells that there are 3 items. When you move focus to the menu item that has a submenu it does not read out how many items are in the submenu (`Menu Item Three, empty group`).

https://user-images.githubusercontent.com/2328042/156378424-4fdea8e6-26c1-4a46-b55e-965664b96245.mov

### Expected behaviour

On select menu item with subitems it should read how many subitems it has.

### CodeSandbox or Storybook URL

https://carbon.sage.com/iframe.html?id=menu--selected&args=&viewMode=story#

### JIRA Ticket (Sage Only)

SBS-28345

### Carbon Version

104.36.0

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by tempertemper** [comment]
This is to do with the HTML/ARIA that has been used. Triggers for dropdowns should be `<button>` elements, and where links are used they [should always have an `href`](https://www.tempertemper.net/blog/links-missing-href-attributes-and-over-engineered-code) attribute (or they're essentially just `<span>`s!).
